pkg/apis/logstash/v1alpha1/logstash_types.go

```diff
@@ -39,6 +39,17 @@ type LogstashSpec struct {
 	// +kubebuilder:validation:Optional
 	ConfigRef *commonv1.ConfigSource `json:"configRef,omitempty"`
 
+	// Pipelines holds the Logstash Pipelines. At most one of [`Pipelines`, `PipelinesRef`] can be specified.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:pruning:PreserveUnknownFields
+	Pipelines []map[string]string `json:"pipelines,omitempty"`
```

Comment:
We might need to rethink this a little - using a `[]map[string]string` here requires the values of numerical values in the yaml to be a string - 
eg

`pipeline.workers: 4` is invalid, and users must specify
`pipeline.workers: "4"`

and requires a flattened yaml structure - 

```
`pipeline:
    id: "demo"
    workers: "4"
```
will also fail due to the requirement for a single string.

From [this](https://github.com/kubernetes-sigs/controller-tools/issues/636) issue, `map[string]interface{}` doesn't appear to be allowed in CRD definitions - there are a couple of workarounds described there that might work for us, and we should investigate. Also, would defining this as a `[]commonv1.Config` work?

Reply:
Many thanks for raising this issue. I changed Pipelines to `[]commonv1.Config`, which can handle integer, string and boolean. I was trying to use `instr.IntOrString` but it fails to handle `true` as string and required `"true"`.

Logstash `pipelines.yml` does not take `pipeline.workers` as a flattened yaml structure 😕 It has to be a `.` in-between, otherwise raises error. Check [here](https://github.com/elastic/logstash/blob/main/logstash-core/lib/logstash/settings.rb#L45-L77) and [here](https://github.com/elastic/logstash/blob/main/logstash-core/lib/logstash/settings.rb#L168).
That's why I have to make [pipelines_config](https://github.com/elastic/cloud-on-k8s/pull/6480/files#diff-afedbf46a40edef9976bb3e3164244742b123e8c4f281242ec886e2769c6882cR22) to not flatten the keys as [canonical_config](https://github.com/elastic/cloud-on-k8s/blob/main/pkg/controller/common/settings/canonical_config.go#L25)